<launch>
    <param name="~frame_id" type="string" value="odom" />

    <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
    <arg name="x_pos" default="-2.0"/>
    <arg name="y_pos" default="0.0"/>
    <arg name="z_pos" default="0.0"/>
    <arg name="world_name" default="pathway.world" doc="name of the world file"/>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find FExT2)/worlds/$(arg world_name)"/>
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>  


    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-urdf -model turtlebot3_burger -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />
 


    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find FExT2)/config/rrt.rviz"/>
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher"/>

    <node pkg="FExT2" type="laser_scan_to_point_cloud_node.py" name="laserscan_to_pointcloud"/>
    <node pkg="octomap_server" type="octomap_server_node" name="octomap" output="screen">
        <remap to="/laserscan_to_pointcloud/converted_pc" from="/cloud_in"/>
    </node>

    <node pkg="FExT2" type="move_to_pt.py" name="move_to_pt"/>
    <node pkg="FExT2" type="frontier_exploration.py" name="frontier_exploration"/>
    
</launch>
